File: backend/src/main/java/com/example/backend/enumerate/StatusEnum.java
package com.example.backend.enumerate;

import com.example.backend.service.GlobalService;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum StatusEnum {
    APROVADO,
    REPROVADO,
    AGUARDANDO;

    @JsonCreator
    public static StatusEnum fromString(String value) {
        String normalized = GlobalService.removeAccents(value).replaceAll("\\s", "").toUpperCase();
        return StatusEnum.valueOf(normalized);
    }

    @JsonValue
    public String toJson() {
        return name();
    }
}

File: frontend/src/app/module/Enumerate/status-enum.ts
export const itensStatus: StatusEnum[] = [];

export class StatusEnum {
  static APROVADO = new StatusEnum('Aprovado', 'APROVADO');
  static REPROVADO = new StatusEnum('Reprovado', 'REPROVADO');
  static AGUARDANDO = new StatusEnum('Aguardando', 'AGUARDANDO');

  label: string;
  value: string;

  private constructor(label: string, value: string) {
    this.label = label;
    this.value = value;
    itensStatus.push(this);
  }
}
